modules/localization/msf/local_tool/local_visualization/engine/visualization_engine.cc

```diff
@@ -176,10 +178,16 @@ void VisualizationEngine::Visualize(
 
   cur_loc_infos_ = loc_infos;
 
-  if (cur_loc_infos_[car_loc_id_].is_valid) {
-    car_pose_ = cur_loc_infos_[car_loc_id_].pose;
-  } else if (!UpdateCarLocId()) {
-    return;
+  if (!UpdateCarLocId(expected_car_loc_id_)) {
+    if (!UpdateCarLocId(car_loc_id_)) {
+      if (!UpdateCarLocId()) {
+        return;
+      }
+    } else {
+      if (expected_car_loc_id_ == loc_info_num_) {
+        expected_car_loc_id_ = car_loc_id_;
+      }
+    }
```

Comment:
Could you reduce these nested conditions? It's not easy to know which branch will be in effect.